app/controllers/stats_controller.rb

```diff
@@ -57,23 +61,19 @@ def create
 
   def download_custom
     report = Report.find(params[:id])
-    data, filename = report.report_details
-
-    if report.report_type.etl?
+    report_data, _ = report.report_details
+    if report.etl?
       return download_waiting(report) unless report.etl_ready?
```

Comment:
if we change `.etl?` we may also want to change `.etl_ready?` to something else. I think just `.ready?` would be fine as the fact that it it was done in the background isn't relevant really.